sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java

```diff
@@ -0,0 +1,386 @@
+     * setMaxPrefetchPageCount(0))
+     * would be when the caller only plans to retrieve just one page - so any prefetched pages
+     * would not be used anyway.
+     *
+     * @param maxPrefetchPageCount the max number of pages that will be prefetched from the backend
+     *                             asynchronously in the background
+     * @return the modified change feed request options.
+     */
+    @Beta(value = Beta.SinceVersion.WHATEVER_NEW_VERSION, warningText =
+        Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosChangeFeedRequestOptions setMaxPrefetchPageCount(int maxPrefetchPageCount) {
+        checkArgument(
+            maxPrefetchPageCount > 0,
+            "Argument 'maxPrefetchCount' must be larger than 0.");
+        this.maxPrefetchPageCount = maxPrefetchPageCount;
+
+        return this;
+    }
+
+    boolean isSplitHandlingDisabled() {
+        return this.isSplitHandlingDisabled;
+    }
+
+    CosmosChangeFeedRequestOptions disableSplitHandling() {
+        this.isSplitHandlingDisabled = true;
+        return this;
+    }
+
+    ChangeFeedMode getMode() {
+        return this.mode;
+    }
+
+    /**
+     * Gets the properties
+     *
+     * @return Map of request options properties
+     */
+    Map<String, Object> getProperties() {
+        return properties;
+    }
+
+    ChangeFeedStartFromInternal getStartFromSettings() {
+        return this.startFromInternal;
+    }
+
+    /**
+     * Creates a new {@link CosmosChangeFeedRequestOptions} instance to start processing
+     * change feed items from the beginning of the change feed
+     *
+     * @param feedRange The {@link FeedRange} that is used to define the scope (entire container,
+     *                  logical partition or subset of a container)
+     * @return a new {@link CosmosChangeFeedRequestOptions} instance
+     */
+    @Beta(value = Beta.SinceVersion.WHATEVER_NEW_VERSION, warningText =
+        Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public static CosmosChangeFeedRequestOptions createForProcessingFromBeginning(FeedRange feedRange) {
+        checkNotNull(feedRange, "Argument 'feedRange' must not be null.");
+
+        return new CosmosChangeFeedRequestOptions(
+            FeedRangeInternal.convert(feedRange),
+            ChangeFeedStartFromInternal.createFromBeginning(),
+            ChangeFeedMode.INCREMENTAL,
+            null);
+    }
+
+    /**
+     * Creates a new {@link CosmosChangeFeedRequestOptions} instance to start processing
+     * change feed items from a previous continuation
+     *
+     * @param continuation The continuation that was retrieved from a previously retrieved
+     *                     FeedResponse
+     * @return a new {@link CosmosChangeFeedRequestOptions} instance
+     */
+    @Beta(value = Beta.SinceVersion.WHATEVER_NEW_VERSION, warningText =
+        Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public static CosmosChangeFeedRequestOptions createForProcessingFromContinuation(
```

Comment:
My understanding is that the "models" classes should only capture properties, any helper functions even if "static" will go in a separate helper class, outside "models" package...

Reply:
As discussed in the Java-sync-up - not considered a hard requirement - and as long as these are factory methods not seen as a problem